### PR TITLE
[NTOSKRNL] include/config.h: Sync it to current code

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Cleanup.

## Proposed changes

```
* SERIALIZE_DBGPRINT: Removed on c15e04c (r18078).
* ENABLE_VALIDATE_POOL: Removed on 5899e14 (r42249).
* TAG_STATISTICS_TRACKING: Removed on 5899e14 (r42249).
* POOL_DEBUG_APIS: Removed on 5899e14 (r42249).
* R_RZ: Removed on 7d21cf7 (r48999).
* R_STACK: Removed on 7d21cf7 (r48999).
* WHOLE_PAGE_ALLOCATIONS: Removed on b431563 (r17306).
+
* _GLOBAL_PAGES_ARE_AWESOME_: Move it here.
```

### CODE REVIEW

**Review by HBelusca — APPROVED** [review]

**Review by ThFabba — APPROVED** [review]

**Comment by Extravert-ir** [comment]
Shouldn't we leave the file for future/current config options?

>

**Comment by SergeGautherie** [comment]
> Shouldn't we leave the file for future/current config options?

Future: It can be re-created if need be.
Current: Which options would you want to add?

**Comment by Extravert-ir** [comment]
There is one switch in the kernel which should belong to this header, I think - `_GLOBAL_PAGES_ARE_AWESOME_`
There are also several ones in subsystems' headers, but for them I'm not sure

**Comment by SergeGautherie** [comment]
> `_GLOBAL_PAGES_ARE_AWESOME_`

Ok: moved around.

> There are also several ones in subsystems' headers, but for them I'm not sure

For another time, then.

File: ntoskrnl/include/ntoskrnl.h [hunk 1]
```diff
@@ -108,13 +108,6 @@
...
-//
-// Switch for enabling global page support
-//
-
-//#define _GLOBAL_PAGES_ARE_AWESOME_
-
-
 /* Internal Headers */
 #include "internal/ntoskrnl.h"
 #include "config.h"
```
Comment (tkreuzer) [inline ntoskrnl/include/ntoskrnl.h:113]: This needs to be imported first.

**Comment by SergeGautherie** [comment]
> This needs to be imported first.

Like this?

**Comment by tkreuzer** [comment]
> 
> 
> > This needs to be imported first.
> 
> Like this?

yes

**Review by tkreuzer — APPROVED** [review]

File: ntoskrnl/include/config.h [hunk 2]
```diff
@@ -1,40 +1,4 @@
...
-    #define R_STACK 0
-
-    /*
-     * Put each block in its own range of pages and position the block at the
-     * end of the range so any accesses beyond the end of block are to invalid
-     * memory locations.
-     */
-    #undef WHOLE_PAGE_ALLOCATIONS
-#endif
+// Enable global page support.
```
Comment (HBelusca) [inline ntoskrnl/include/config.h:3]:
```suggestion
/* Enable global page support. */
```

**Pull request merged**